ta/pkcs11/src/pkcs11_attributes.c

```diff
@@ -2478,7 +2484,34 @@ static enum pkcs11_rc set_secret_key_data(struct obj_attrs **head, void *data,
 	if (rc != PKCS11_CKR_OK && rc != PKCS11_RV_NOT_FOUND)
 		return PKCS11_CKR_GENERAL_ERROR;
 
-	return add_attribute(head, PKCS11_CKA_VALUE, data, key_length);
+	rc = add_attribute(head, PKCS11_CKA_VALUE, data, key_length);
+	if (rc != PKCS11_CKR_OK)
+		return rc;
+
+	/* Remove the default empty check value attribute if found */
+	rc = remove_empty_attribute(head, PKCS11_CKA_CHECK_VALUE);
+	if (rc != PKCS11_CKR_OK && rc != PKCS11_RV_NOT_FOUND)
+		return  PKCS11_CKR_GENERAL_ERROR;
+
+	/* Get key check value size */
+	rc = aes_get_check_value_size(data, key_length, &kcv_len);
+	if (rc)
+		return rc;
+
+	kcv = TEE_Malloc(kcv_len, TEE_MALLOC_FILL_ZERO);
+	if (!kcv)
+		return PKCS11_CKR_DEVICE_MEMORY;
+
+	/* Compute and add the key check sum value*/
+	rc = aes_compute_check_value(data, key_length, kcv, &kcv_len);
+	if (rc != PKCS11_CKR_OK)
+		return  PKCS11_CKR_GENERAL_ERROR;
```

Comment:
Need to balance `TEE_Malloc()` above in case of error.

Could this return `rc` instead of `PKCS11_CKR_GENERAL_ERROR`, as done in `generate_random_key_value()`?

Suggestion:
```c
	rc = aes_compute_check_value(data, key_length, kcv, &kcv_len);
	if (rc == PKCS11_CKR_OK) 
			rc = add_attribute(head, PKCS11_CKA_CHECK_VALUE, kcv, kcv_len);

	TEE_Free(kcv);

	return rc;
```